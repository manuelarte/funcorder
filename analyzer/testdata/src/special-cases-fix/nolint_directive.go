package special_cases_fix

//nolint:lll // test directive
func NewNoLLLDirective() *NoLLLDirective { // want `constructor "NewNoLLLDirective" for struct "NoLLLDirective" should be placed after the struct declaration`
	return &NoLLLDirective{}
}

type NoLLLDirective struct{}

// NewNoLLLDirectiveWithDoc constructor for NoLLLDirectiveWithDoc
//
//nolint:lll // test directive
func NewNoLLLDirectiveWithDoc() *NoLLLDirectiveWithDoc { // want `constructor "NewNoLLLDirectiveWithDoc" for struct "NoLLLDirectiveWithDoc" should be placed after the struct declaration`
	return &NoLLLDirectiveWithDoc{}
}

// NoLLLDirectiveWithDoc with docs.
type NoLLLDirectiveWithDoc struct{}
