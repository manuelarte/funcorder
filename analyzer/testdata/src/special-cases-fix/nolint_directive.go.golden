package special_cases_fix

type NoLLLDirective struct{}

//nolint:lll // test directive
func NewNoLLLDirective() *NoLLLDirective {
	return &NoLLLDirective{}
}

// NoLLLDirectiveWithDoc with docs.
type NoLLLDirectiveWithDoc struct{}

// NewNoLLLDirectiveWithDoc constructor for NoLLLDirectiveWithDoc
//
//nolint:lll // test directive
func NewNoLLLDirectiveWithDoc() *NoLLLDirectiveWithDoc {
	return &NoLLLDirectiveWithDoc{}
}
